File: doc-examples/jdbc-example-java/src/main/java/example/Address.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity(value = "TBL_ADDRESS", alias = "a")
public record Address (
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    Long id,
    String street,
    String city
) {}

File: doc-examples/jdbc-example-python/src/main/python/example/Address.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity


@MappedEntity(value="TBL_ADDRESS", alias="a")
@dataclass
class Address:
    street: str
    city: str
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
